modules/installation-aws-upload-custom-rhcos-ami.adoc

```diff
@@ -0,0 +1,141 @@
+be the highest version that is less than or equal to the {product-title} version
+you are installing.
+* Download the AWS CLI and install it on your computer. See
+link:https://docs.aws.amazon.com/cli/latest/userguide/install-bundle.html[Install the AWS CLI Using the Bundled Installer (Linux, macOS, or Unix)].
+
+.Procedure
+
+. Export your AWS profile as an environment variable:
++
+[source,terminal]
+----
+$ export AWS_PROFILE=<aws_profile> <1>
+----
+<1> The AWS profile name that holds your AWS credentials, like `govcloud`.
+
+. Export the region to associate with your custom AMI as an environment
+variable:
++
+[source,terminal]
+----
+$ export AWS_DEFAULT_REGION=<aws_region> <1>
+----
+<1> The AWS region, like `us-gov-east-1`.
+
+. Export the version of {op-system} you uploaded to Amazon S3 as an environment
+variable:
++
+[source,terminal]
+----
+$ export RHCOS_VERSION=<version> <1>
+----
+<1> The {op-system} VMDK version, like `4.6.0`.
+
+. Export the Amazon S3 bucket name as an environment variable:
++
+[source,terminal]
+----
+$ export VMIMPORT_BUCKET_NAME=<s3_bucket_name>
+----
+
+. Create the `containers.json` file and define the credentials for your
+{op-system} VMDK file:
++
+[source,terminal]
+----
+$ cat <<EOF > containers.json
+{
+   "Description": "rhcos-${RHCOS_VERSION}-x86_64-aws.x86_64",
+   "Format": "vmdk",
+   "UserBucket": {
+      "S3Bucket": "${VMIMPORT_BUCKET_NAME}",
+      "S3Key": "rhcos-${RHCOS_VERSION}-x86_64-aws.x86_64.vmdk"
+   }
+}
+EOF
+----
+
+. Import the {op-system} disk as an Amazon EBS snapshot:
++
+[source,terminal]
+----
+$ aws ec2 import-snapshot --region ${AWS_DEFAULT_REGION} \
+     --description "<description>" \ <1>
+     --disk-container <file_path>/containers.json <2>
+----
+<1> The description of your {op-system} disk being imported, like
+`rhcos-${RHCOS_VERSION}-x86_64-aws.x86_64`.
+<2> The file path to the JSON file describing your {op-system} disk. The JSON
+file should contain the credentials for your Amazon S3 bucket and key.
+
+. Check the status of the image import:
++
+[source,terminal]
+----
+$ watch -n 5 aws ec2 describe-import-snapshot-tasks --region ${AWS_DEFAULT_REGION}
+----
++
+.Example output
+[source,terminal]
+----
+{
+    "ImportSnapshotTasks": [
+        {
+            "Description": "rhcos-4.6.0-x86_64-aws.x86_64",
+            "ImportTaskId": "import-snap-fh6i8uil",
+            "SnapshotTaskDetail": {
+                "Description": "rhcos-4.6.0-x86_64-aws.x86_64",
+                "DiskImageSize": 819056640.0,
+                "Format": "VMDK",
+                "SnapshotId": "snap-06331325870076318",
+                "Status": "completed",
+                "UserBucket": {
+                    "S3Bucket": "external-images",
+                    "S3Key": "rhcos-4.6.0-x86_64-aws.x86_64.vmdk"
+                }
+            }
+        }
+    ]
+}
+----
++
+Copy the `SnapshotId` to register the image.
```

Comment:
I realize my version of the details here didn't include this either, but maybe it's worth adding a command using `jq` to extract the value? I know in other parts of the OpenShift docs, the `jq` command is used so there seems to be precedence for it.

Reply:
Although `jq` is used in some of our installer docs, we're avoiding the addition of any new` jq` commands since it isn't an available package by default in RHEL.. The existing ones will be replaced/removed at some point in the future.

Reply:
That makes sense to me, thanks!